sklearn/utils/__init__.py

```diff
@@ -983,6 +984,25 @@ def get_chunk_n_rows(row_bytes, *, max_n_rows=None, working_memory=None):
     return chunk_n_rows
 
 
+def is_pd_na(x):
+    """Test if x is pd.NA.
+
+    Parameters
+    ----------
+    x : any type
+
+    Returns
+    -------
+    boolean
+    """
+    with suppress(ImportError, AttributeError):
```

Comment:
I personally prefer to check `x is pandas.NA` directly, because `NA` scalar is "still in experimental and its behavior can change without warning": https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html#missing-data-na.